hbase-rest/src/main/java/org/apache/hadoop/hbase/rest/model/ScannerModel.java

```diff
@@ -121,8 +121,10 @@ public class ScannerModel implements ProtobufMessageHandler, Serializable {
   private boolean cacheBlocks = true;
   private int limit = -1;
 
+  @JsonInclude(value = JsonInclude.Include.CUSTOM, valueFilter = IncludeStartRowFIlter.class)
```

Comment:
Why do we need the filter for this and not for the stopRow ?
Wouldn't JsonInclude.Include.NON_DEFAULT work here the same ?

Reply:
Also, a Unit test to check this serialization behaviour would be helpful.

Reply:
To make sure that it survives any later refactors, etc.

Reply:
Looks like there is already a test, it just needs to be updated.

Reply:
> Why do we need the filter for this and not for the stopRow ?

JsonInclude.Include.NON_DEFAULT --> this annotation serialize the attribute only if the attribute value is different from it's default value , ex: for boolean variables if the value is false(default) it will not get serialized, if the value is true it gets serialized.

Since includeStopRow is false by default we can use this annotation, where as includeStartRow is true by default, so we can't use this annotation because when the value is set to false this attribute will not get serialized and it will be giving wrong results in the response of scan request.

Reply:
> Looks like there is already a test, it just needs to be updated.

updated